dropshot/src/extractor/header.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2025 Oxide Computer Company
+
+//! Header extractor
+
+use std::collections::BTreeMap;
+
+use async_trait::async_trait;
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
+
+use crate::{
+    from_map::from_map, ApiEndpointBodyContentType,
+    ApiEndpointParameterLocation, HttpError, RequestContext, RequestInfo,
+    ServerContext,
+};
+
+use super::{metadata::get_metadata, ExtractorMetadata, SharedExtractor};
+
+pub struct Header<HeaderType: DeserializeOwned + JsonSchema + Send + Sync> {
```

Comment:
This is part of the public-facing API; it might be worth documenting (or including a small example)

Reply:
Good point; thanks